bigc/resources/orders.py

```diff
@@ -72,3 +73,40 @@ def all_shipping_addresses(self, order_id: int) -> Iterator[dict]:
     def get_shipping_address(self, order_id: int, address_id: int) -> dict:
         """Get a specific shipping address in an order by ID"""
         return self._api.v2.get(f'/orders/{order_id}/shipping_addresses/{address_id}')
+
+    def all_shipments(self, order_id: int) -> Iterator[dict]:
+        """Returns all shipments for a specified order"""
+
+        return self._api.v2.get_many(f'/orders/{order_id}/shipments')
+
+    def create_shipment(
+            self,
+            order_id: int,
+            order_address_id: int,
+            items: list[dict],
+            *,
+            tracking_number: Optional[str] = None,
+            shipping_method: Optional[BigCommerceShippingMethod] = None,
+            shipping_provider: Optional[BigCommerceShippingProvider] = None,
+            tracking_carrier: Optional[str] = None,
+            comments: Optional[str] = None,
+            **kwargs,
+    ) -> dict:
```

Comment:
IMO this is overcomplicating it. Trying to reproduce all of BigCommerce's fields in this library is just going to be way too much work to maintain.

Generally the pattern we have used so far is:

* Any "path" args (e.g. `order_id`)
* `*` to make sure everything else is a keyword arg
* Required fields (e.g. `order_address_id`)
* `**kwargs` for all optional fields

This is how I would expect this signature to look based on the docs:

```py
def create_shipment(self, order_id: int, *, order_address_id: int, items: list[dict], **kwargs) -> dict:
```

Reply:
https://github.com/MedShift/bigc/pull/25/commits/bdfb8fb7cdc7d4338022155075cb983aa2a9af02